command-parser.js

```diff
@@ -233,199 +456,20 @@ var parse = exports.parse = function (message, room, user, connection, levelsDee
-					if (!message) return false;
-					if (!user.can('broadcast', null, room)) {
-						connection.sendTo(room, "You need to be voiced to broadcast this command's information.");
-						connection.sendTo(room, "To see it for yourself, use: /" + message.substr(1));
-						return false;
-					}
-
-					// broadcast cooldown
-					var normalized = message.toLowerCase().replace(/[^a-z0-9\s!,]/g, '');
-					if (room.lastBroadcast === normalized &&
-							room.lastBroadcastTime >= Date.now() - BROADCAST_COOLDOWN) {
-						connection.sendTo(room, "You can't broadcast this because it was just broadcast.");
-						return false;
-					}
-					this.add('|c|' + user.getIdentity(room.id) + '|' + (suppressMessage || message));
-					room.lastBroadcast = normalized;
-					room.lastBroadcastTime = Date.now();
-
-					this.broadcasting = true;
-				}
-				return true;
-			},
-			parse: function (message, inNamespace) {
-				if (inNamespace && (message[0] === '/' || message[0] === '!')) {
-					message = message[0] + namespaces.concat(message.slice(1)).join(" ");
-				}
-				return parse(message, room, user, connection, levelsDeep + 1);
-			},
-			canTalk: function (message, relevantRoom, targetUser) {
-				var innerRoom = (relevantRoom !== undefined) ? relevantRoom : room;
-				return canTalk(user, innerRoom, connection, message, targetUser);
-			},
-			canHTML: function (html) {
-				html = '' + (html || '');
-				var images = html.match(/<img\b[^<>]*/ig);
-				if (!images) return true;
-				for (var i = 0; i < images.length; i++) {
-					if (!/width=([0-9]+|"[0-9]+")/i.test(images[i]) || !/height=([0-9]+|"[0-9]+")/i.test(images[i])) {
-						this.sendReply('All images must have a width and height attribute');
-						return false;
-					}
-				}
-				if (/>here.?</i.test(html) || /click here/i.test(html)) {
-					this.sendReply('Do not use "click here"');
-					return false;
-				}
-
-				// check for mismatched tags
-				var tags = html.toLowerCase().match(/<\/?(div|a|button|b|i|u|center|font)\b/g);
-				if (tags) {
-					var stack = [];
-					for (var i = 0; i < tags.length; i++) {
-						var tag = tags[i];
-						if (tag.charAt(1) === '/') {
-							if (!stack.length) {
-								this.sendReply("Extraneous </" + tag.substr(2) + "> without an opening tag.");
-								return false;
-							}
-							if (tag.substr(2) !== stack.pop()) {
-								this.sendReply("Missing </" + tag.substr(2) + "> or it's in the wrong place.");
-								return false;
-							}
-						} else {
-							stack.push(tag.substr(1));
-						}
-					}
-					if (stack.length) {
-						this.sendReply("Missing </" + stack.pop() + ">.");
-						return false;
-					}
-				}
+		var context = new CommandContext({
+			target: target, room: room, user: user, connection: connection, cmd: cmd, message: message,
+			namespaces: namespaces, commands: currentCommands, originalMessage: originalMessage,
+			cmdToken: cmdToken, levelsDeep: levelsDeep, _cmdMain: _cmdMain
```

Comment:
I'm not sure if there's a need for `Context#run()` with no arguments. What's the use case?